src/core/MOM_stoch_eos.F90

```diff
@@ -0,0 +1,212 @@
+!> Provides the ocean stochastic equation of state
+module MOM_stoch_eos
+! This file is part of MOM6. See LICENSE.md for the license.
+use MOM_grid,            only : ocean_grid_type
+use MOM_hor_index,       only : hor_index_type
+use MOM_file_parser,     only : get_param, param_file_type
+use MOM_random,          only : PRNG,random_2d_constructor,random_2d_norm
+use MOM_time_manager,    only : time_type
+use MOM_io,              only : vardesc, var_desc
+use MOM_restart,         only : MOM_restart_CS,is_new_run
+use MOM_diag_mediator,   only : register_diag_field,post_data,diag_ctrl,safe_alloc_ptr
+use MOM_variables,       only : thermo_var_ptrs
+use MOM_verticalGrid,    only : verticalGrid_type
+use MOM_restart,         only : register_restart_field
+use MOM_isopycnal_slopes,only : vert_fill_TS
+!use random_numbers_mod, only : getRandomNumbers,initializeRandomNumberStream,randomNumberStream
+
+implicit none
+#include <MOM_memory.h>
+
+public MOM_stoch_eos_init
+public MOM_stoch_eos_run
+public MOM_calc_varT
+
+real,private ALLOCABLE_, dimension(NIMEM_,NJMEM_) :: l2_inv
+                        !< One over sum of the T cell side side lengths squared
+real,private ALLOCABLE_, dimension(NIMEM_,NJMEM_) :: rgauss !< nondimensional random Gaussian
+real, parameter,private :: tfac=0.27 !< Nondimensional decorrelation time factor, ~1/3.7
+real, parameter,private :: amplitude=0.624499 !< Nondimensional std dev of Gaussian
+integer        ,private :: seed !< PRNG seed
+type(PRNG)  ::  rn_CS !< PRNG control structure
+
+!> Describes parameters of the stochastic component of the EOS
+!! correction, described in Stanley et al. JAMES 2020.
+type, public :: MOM_stoch_eos_CS
+  real,public  ALLOCABLE_, dimension(NIMEM_,NJMEM_) :: pattern
+                    !< Random pattern for stochastic EOS
+  real ALLOCABLE_, dimension(NIMEM_,NJMEM_) :: phi
+                  !< temporal correlation stochastic EOS (deugging)
```

Comment:
Please add a description of the units for all real variables, even if this is `[various]` for fields that are reused with different units in different contexts or `[nondim]` for nondimensional variables.  Whenever appropriate, these units should also be logged in the get_param calls that are used to set variables.